packages/fxa-payments-server/src/components/PaymentErrorView/index.tsx

```diff
@@ -70,39 +83,74 @@ export const PaymentErrorView = ({
         return manageSubButtonFn(() => navigate('/subscriptions'));
       case 'iap_already_subscribed':
         return manageSubButtonFn(actionFn);
+      case 'iap_upgrade_contact_support':
+        return iapStopUpgradeSubButtonFn();
       default:
         return retryButtonFn(actionFn);
     }
   };
 
+  // We want to know which layout to display
+  const screenType = () => {
+    switch (error?.code) {
+      case 'iap_upgrade_contact_support':
+        return 'iaperrorupgrade';
+      case 'iap_already_subscribed':
+      default:
+        return 'error';
+    }
+  };
+
   const title = subscriptionTitle ?? (
-    <SubscriptionTitle screenType="error" className={className} />
+    <SubscriptionTitle screenType={screenType()} className={className} />
   );
 
   const productName = plan.product_name;
 
+  // Returns an array of localized errors, opening the door for multiline
+  // without having to retranslate whole blocks for one word or one line changes.
+  // Currently hardcodes l10n ids for iap subscription upgrade roadblock error
+  // as per https://mozilla-hub.atlassian.net/browse/FXA-5480 (full refactor out of scope)
+  // Issue filed to refactor and dynamically add ids: https://mozilla-hub.atlassian.net/browse/FXA-6100
+  const PaymentErrorMessage = () => {
```

Comment:
I believe you'd mentioned we wanted to make this logic more generic in future. Do we already have a Jira ticket for that?

Reply:
Yep! I left a comment in the file but should be address in FXA-6100 🙂 .